gcc/config/riscv/rvp.md

```diff
@@ -0,0 +1,7058 @@
+  [(match_operand:SI 0 "register_operand" "")
+   (match_operand:SI 1 "register_operand" "")
+   (match_operand:V2HI 2 "register_operand" "")
+   (match_operand:V2HI 3 "register_operand" "")]
+  "TARGET_ZPN"
+{
+  emit_insn (gen_kdm_internal (operands[0], operands[2], operands[3],
+				GEN_INT (0), GEN_INT (1), operands[1]));
+  DONE;
+}
+[(set_attr "type" "dsp")])
+
+(define_expand "kdmtt"
+  [(match_operand:SI 0 "register_operand" "")
+   (match_operand:SI 1 "register_operand" "")
+   (match_operand:V2HI 2 "register_operand" "")
+   (match_operand:V2HI 3 "register_operand" "")]
+  "TARGET_ZPN"
+{
+  emit_insn (gen_kdm_internal (operands[0], operands[2], operands[3],
+				GEN_INT (1), GEN_INT (1), operands[1]));
+  DONE;
+}
+[(set_attr "type" "dsp")])
+
+(define_insn "kdm_internal"
+  [(set (match_operand:SI 0 "register_operand"                      "=   r,   r,   r,   r")
+	(ss_plus:SI
+	  (ashift:SI
+	    (mult:SI
+	      (sign_extend:SI
+		(vec_select:HI
+		  (match_operand:V2HI 1 "register_operand"          "   r,   r,   r,   r")
+		  (parallel [(match_operand:SI 3 "imm_0_1_operand"  " C00, C00, C01, C01")])))
+	      (sign_extend:SI
+		(vec_select:HI
+		  (match_operand:V2HI 2 "register_operand"          "   r,   r,   r,   r")
+		  (parallel [(match_operand:SI 4 "imm_0_1_operand"  " C00, C01, C01, C00")]))))
+	    (const_int 1))
+	  (match_operand:SI 5 "register_operand"                    "   0,   0,   0,   0")))]
+  "TARGET_ZPN"
+  "@
+   kdmbb\t%0, %1, %2
+   kdmbt\t%0, %1, %2
+   kdmtt\t%0, %1, %2
+   kdmbt\t%0, %2, %1"
+  [(set_attr "type" "dsp")
+   (set_attr "mode" "SI")])
+
+;; for RV64 vector intrinsic
+;; in RV64, int32_t __rv__v_kdmtt(int16x4_t a, int16x4_t b);
+(define_expand "v64_kdmbb"
+  [(match_operand:SI 0 "register_operand" "")
+   (match_operand:SI 1 "register_operand" "")
+   (match_operand:V4HI 2 "register_operand" "")
+   (match_operand:V4HI 3 "register_operand" "")]
+  "TARGET_ZPN && TARGET_64BIT"
+{
+  emit_insn (gen_kdm64_internal (operands[0], operands[2], operands[3],
+				GEN_INT (0), GEN_INT (0), operands[1]));
+  DONE;
+}
+[(set_attr "type" "dsp")])
+
+(define_expand "v64_kdmbt"
+  [(match_operand:SI 0 "register_operand" "")
+   (match_operand:SI 1 "register_operand" "")
+   (match_operand:V2HI 2 "register_operand" "")
+   (match_operand:V2HI 3 "register_operand" "")]
+  "TARGET_ZPN && TARGET_64BIT"
+{
+  emit_insn (gen_kdm64_internal (operands[0], operands[2], operands[3],
+				GEN_INT (0), GEN_INT (1), operands[1]));
+  DONE;
+}
+[(set_attr "type" "dsp")])
+
+(define_expand "v64_kdmtt"
+  [(match_operand:SI 0 "register_operand" "")
+   (match_operand:SI 1 "register_operand" "")
+   (match_operand:V2HI 2 "register_operand" "")
+   (match_operand:V2HI 3 "register_operand" "")]
+  "TARGET_ZPN && TARGET_64BIT"
+{
+  emit_insn (gen_kdm64_internal (operands[0], operands[2], operands[3],
+				GEN_INT (1), GEN_INT (1), operands[1]));
+  DONE;
+}
+[(set_attr "type" "dsp")])
+
+(define_insn "kdm64_internal"
```

Comment:
Wrong descriptions.
Check with
```
int main() {
    return 0
        || (__rv__kdmbb(0x80008000, 0x80008000) - 0x7FFFFFFF)
        || (__rv__kdmbt(0x80008000, 0x80008000) - 0x7FFFFFFF)
        || (__rv__kdmtt(0x80008000, 0x80008000) - 0x7FFFFFFF)
    ;
}
```

Reply:
Thanks, and actually `kdm64_internal` was unused for intrinsics.

I just added a patch for it to replace the previous unspec patterns, and I will update an execution version of test cases, and I think it would be helpful for checking the correctness. Thank you again.